### PR TITLE
- Reuse NetBios connection buffer

### PR DESCRIPTION
**Pull request by itsKontra**
- Reduce buffer size to half allocation

### CODE REVIEW

**Comment by itsKontra** [comment]
Pullrequest for #202

File: SMBLibrary/Client/SMB2Client.cs [hunk 1]
```diff
@@ -21,9 +21,10 @@ public class SMB2Client : ISMBClient
         public static readonly int NetBiosOverTCPPort = 139;
         public static readonly int DirectTCPPort = 445;
 
-        public static readonly uint ClientMaxTransactSize = 1048576;
-        public static readonly uint ClientMaxReadSize = 1048576;
-        public static readonly uint ClientMaxWriteSize = 1048576;
+        private static readonly uint RawDataBufferSize = 1048576 - 256 - SessionPacket.HeaderLength;
```
Comment (itsKontra) [inline SMBLibrary/Client/SMB2Client.cs:24]: ArrayPool buffer size is of power of 2. 
Here we reduce the buffer size by MaxTransactSize  and HeaderLength. 
Buffer has now 1Mib instead of 2Mib per request.

File: SMBLibrary/Client/ConnectionState.cs [hunk 2]
```diff
@@ -4,41 +4,34 @@
...
         }
 
-        public Socket ClientSocket
-        {
-            get
-            {
-                return m_clientSocket;
-            }
-        }
+        public Socket ClientSocket => m_clientSocket;
```
Comment (TalAloni) [inline SMBLibrary/Client/ConnectionState.cs:25]: How are those changes related to buffer reuse?
It's best to refrain from making unrelated changes in a single PR that shift focus from what you're trying to achieve

**Comment by SydoxX** [comment]
Any news about this issue? I'd be happy to help as I also need this fix :)

**Comment by TalAloni** [comment]
There was a proposed workaround (See #202),
I wanted to take a deeper look into how this could be implemented in the cleanest way possible, however that took more than I expected and I didn't yet manage to finish the work (mostly due to summer vacation which took a lot of planning). I'll try to get around to it.

**Comment by JonasReichhardt** [comment]
I want to mention I also have problems with RAM usage. Is there any news on that topic?

**Comment by TalAloni** [comment]
Sorry, will try to get to it soon

**Comment by TalAloni** [comment]
Hi, sorry for the delay.
Thanks for the submission, after reviewing the code, I opted to use a similar but different approach,
to save time I've already went ahead and pushed the changes, please review and provide feedback.

**Comment by TalAloni** [comment]
Thanks again for the detailed information provided here, RAM consumption has been reduced in v1.5.1

**Comment by SydoxX** [comment]
Works! Thank you for merging :)
